addons/event/models/event_registration.py

```diff
@@ -83,25 +98,34 @@ def _prepare_attendee_values(self, registration):
         data.update({key: value for key, value in registration.items() if key in self._fields})
         return data
 
+    def name_get(self):
+        ret_list = []
+        for registration in self:
+            if not registration.partner_id or not registration.partner_id.name:
+                name = registration.name
+            elif not registration.name or registration.name == registration.partner_id.name:
+                name = registration.partner_id.name
+            else:
+                name = registration.partner_id.name + ', ' + registration.name
+            ret_list.append((registration.id, name))
+        return ret_list
+
     def do_draft(self):
         self.write({'state': 'draft'})
 
     def confirm_registration(self):
         self.write({'state': 'open'})
 
-        # auto-trigger after_sub (on subscribe) mail schedulers, if needed
-        onsubscribe_schedulers = self.event_id.event_mail_ids.filtered(
-            lambda s: s.interval_type == 'after_sub')
-        onsubscribe_schedulers.execute()
-
-    def button_reg_close(self):
+    def action_reg_close(self):
         """ Close Registration """
-        for registration in self:
-            registration.write({'state': 'done', 'date_closed': fields.Datetime.now()})
+        self.write({'state': 'done'})
 
-    def button_reg_cancel(self):
+    def action_reg_cancel(self):
         self.write({'state': 'cancel'})
 
+    def action_reg_confirm(self):
```

Comment:
There is a mismatch here if I am not wrong: action_reg_confirm makes as done (and does not seem to be used).